packages/react-reconciler/src/ReactFiberBeginWork.js

```diff
@@ -1485,14 +1507,19 @@ function mountIndeterminateComponent(
         debugRenderPhaseSideEffectsForStrictMode &&
         workInProgress.mode & StrictMode
       ) {
-        value = renderWithHooks(
-          null,
-          workInProgress,
-          Component,
-          props,
-          context,
-          renderExpirationTime,
-        );
+        disableLogs();
+        try {
+          value = renderWithHooks(
```

Comment:
I could foresee some confusion if the logs don't correspond to the value that is returned. It usually doesn't matter because it's supposed to be pure, but in the case where it's not, you might use `console` to debug it. So maybe they should be disabled for the first pass instead of the second.

Reply:
So I was actually surprised that we used the second value. I was thinking we should ignore the second value like we do for class constructors.

Reply:
In my other PR's replaying I have to ignore it because I force an error inside the function. Similarly the react-debug-tools for Hooks debugging it's just replayed stand-alone.

Doesn't necessarily mean we have to do the same here. I guess that's how I look at these.

Reply:
It's intentional because the children are bound to the work-in-progress fiber via the `queue` object via the `dispatch` method, and `renderWithHooks` mutates the fiber.

It was a very confusing bug when we found it :D

Reply:
This one: https://github.com/facebook/react/pull/14698

Reply:
I guess that's why I had to update that test to get the right dispatcher.

Reply:
I could go either way on if the second render should process the update queue and even leave any side-effects. :)

Reply:
Maybe the second render should have a different dispatcher that warns if you call setState again since you shouldn't call setState again for the same input.

Reply:
> I could go either way on if the second render should process the update queue and even leave any side-effects  :)

It can work but it needs to include internal mutations/side-effects too. We'd have to make it so that `renderWithHooks` does not mutate any of the `fiber` fields. Or stash the result of the first pass (`memoizedState`, `expirationTime`, `effectTag`) before doing the second pass and then restore them.

Reply:
> Maybe the second render should have a different dispatcher that warns if you call setState again since you shouldn't call setState again for the same input.

Oh I didn't see that. Yeah I think that would work. Just call the function again with a special dispatcher that doesn't mutate anything.

Reply:
> Maybe the second render should have a different dispatcher that warns if you call setState again since you shouldn't call setState again for the same input.

I might be misunderstanding this comment, but if the second call has state updates already applied, then we might miss out on triggering the observable side effects in the first place (e.g. like something that happens only when props/state meet a certain criteria).